docs/csharp/tutorials/string-interpolation.md

```diff
@@ -0,0 +1,113 @@
+﻿---
+title: String Interpolation | C#
+description: Learn how string interpolation works in C# 6
+keywords: .NET, .NET Core, C#, string
+author: mgroves
+manager: BillWagner
+ms.date: 12/09/2016
+ms.topic: article
+ms.prod: .net-core
+ms.technology: .net-core-technologies
+ms.devlang: csharp
+ms.assetid: f8806f6b-3ac7-4ee6-9b3e-c524d5301ae9
+---
+
+# String Interpolation in C#
+
+## Introduction
+
+- string.format
+- other languages
+- how it looks in c# now
+
+## Prerequisites
+You’ll need to setup your machine to run .NET core. You can find the
+installation instructions on the [.NET Core](https://www.microsoft.com/net/core)
+page.
+You can run this application on Windows, Ubuntu Linux, macOS or in a Docker container. 
+You’ll need to install your favorite code editor. The descriptions below
+use [Visual Studio Code](https://code.visualstudio.com/) which is an open
+source, cross platform editor. However, you can use whatever tools you are
+comfortable with.
+
+## Create the Application
+
+Now that you've installed all the tools, create a new .NET Core
+application. To use the command line generator, execute the following command in your favorite shell:
+
+`dotnet new`
+
+This command will create a barebones .NET core project with `project.json` and `Program.cs` files. You will need to execute `dotnet restore` to restore the dependencies needed to compile this project.
+
+To execute the program, use `dotnet run`. You should see "Hello, World" output to the console.
+
+## Intro to String Interpolation
+
+With `String.Format`, you would specify "placeholders" in a string that would be replaced by the parameters following the string. For instance:
+
+[!code-csharp[String.Format example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#StringFormatExample)]  
+
+That will output "My name is Matt Groves".
+
+In C# 6, Instead of using String.Format, you can tell C# that a string is interpolated by prepending it with the `$` symbol, and then using the variables directly in the string. For instance:
+
+[!code-csharp[Interpolation example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#InterpolationExample)]  
+
+You don't have to use just variables. You can use any expression within the brackets. For instance:
+
+[!code-csharp[Interpolation expression example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#InterpolationExpressionExample)]  
+
+Which would output:
+
+```
+This is line number 1
+This is line number 2
+This is line number 3
+This is line number 4
+This is line number 5
+```
+
+## How string interpolation works
+
+Behind the scenes, this string interpolation syntax gets translated into String.Format by the compiler. So, you can do the same type of stuff you've done before with String.Format.
+
+For instance, you can add padding and numeric formatting:
```

Comment:
Let's add a link to the format strings doc [https://msdn.microsoft.com/en-us/library/dwhawy9k(v=vs.110).aspx](https://msdn.microsoft.com/en-us/library/dwhawy9k(v=vs.110).aspx) here to explain how that works.

(We'll catch and update those links when that section in MSDN is migrated).